### PR TITLE
Add getters to ServiceBusProcessorClient to get names of SB entity

### PR DESCRIPTION
**Pull request by srnagar**
# Description

This PR adds getter methods to `ServiceBusProcessorClient` to get queue, topic and subscription names associated with the processor.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java [hunk 1]
```diff
@@ -135,17 +135,24 @@
...
+    private final String subscriptionName;
     private ScheduledExecutorService scheduledExecutor;
 
     /**
      * Constructor to create a sessions-enabled processor.
      *
      * @param sessionReceiverBuilder The session processor builder to create new instances of async clients.
+     * @param queueName The name of the queue this session processor is associated with.
+     * @param topicName The name of the topic this session processor is associated with.
+     * @param queueName The name of the subscription this session processor is associated with.
```
Comment (saragluna) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java:149]: subscriptionName

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java [hunk 2]
```diff
@@ -157,17 +164,24 @@
...
+        this.subscriptionName = subscriptionName;
     }
 
     /**
      * Constructor to create a processor.
      *
      * @param receiverBuilder The processor builder to create new instances of async clients.
+     * @param queueName The name of the queue this session processor is associated with.
+     * @param topicName The name of the topic this session processor is associated with.
+     * @param queueName The name of the subscription this session processor is associated with.
```
Comment (saragluna) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java:178]: subscriptionName

Comment (saragluna) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java:178]: we should remove `session` here.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]
LGTM.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**